### PR TITLE
CORE-10444 : Changes to create interop API

### PR DESCRIPTION
**Pull request by SaurabhR3**
1. Created interop rest resource with two api 1) Get interop groups and 2) Create interop identity
2. Fixed issue with package info and bnd test
3. Fixed OpenAPICompatibilityTest
4. Updated SwaggerBaseline.json

I have kept some of the code commented as Julia needs this for API development

### CODE REVIEW

**Review by julia-filipczak — APPROVED** [review]

**Review by jmacmahonr3 — APPROVED** [review]
LGTM

File: processors/rest-processor/src/integrationTest/kotlin/net/corda/processors/rest/OpenApiCompatibilityTest.kt [hunk 1]
```diff
@@ -51,7 +52,8 @@ class OpenApiCompatibilityTest {
             FlowRestResource::class.java, // Flow
             FlowClassRestResource::class.java, // Flow
             CpiUploadRestResource::class.java, // Packaging
-            VirtualNodeRestResource::class.java, // Packaging
+            VirtualNodeRestResource::class.java, // Packaging,
+            InteropRestResource::class.java,
```
Comment (yarivus) [inline processors/rest-processor/src/integrationTest/kotlin/net/corda/processors/rest/OpenApiCompatibilityTest.kt]: for consistency add a comment too ;)

File: components/interop/interop-rest-service-impl/src/main/kotlin/net/corda/interop/rest/impl/v1/InteropRestResourceImpl.kt [hunk 2]
```diff
@@ -0,0 +1,113 @@
...
+    override val targetInterface: Class<InteropRestResource> = InteropRestResource::class.java
+    override fun getInterOpGroups(): List<UUID> {
+        return listOf(UUID.randomUUID())
+    }
+
+    override fun createInterOpIdentity(x500Name: String, groupId: String): ResponseEntity<AsyncResponse> {
+        TODO("Not yet implemented")
+    }
+
+//    override fun createInterOpIdentity(x500Name: String, groupId: UUID): ResponseEntity<AsyncResponse> {
```
Comment (yarivus) [inline components/interop/interop-rest-service-impl/src/main/kotlin/net/corda/interop/rest/impl/v1/InteropRestResourceImpl.kt]: remove commented out code

**Review by yarivus — CHANGES_REQUESTED** [review]

File: processors/db-processor/src/main/kotlin/net/corda/processors/db/internal/DBProcessorImpl.kt [hunk 3]
```diff
@@ -113,6 +115,11 @@ class DBProcessorImpl @Activate constructor(
     private val membershipGroupPolicyValidator: MembershipGroupPolicyValidator,
     @Reference(service = AllowedCertificatesReaderWriterService::class)
     private val allowedCertificatesReaderWriterService: AllowedCertificatesReaderWriterService,
+    // TODO: Temporarily added here, reevaluate this before merge
```
Comment (yarivus) [inline processors/db-processor/src/main/kotlin/net/corda/processors/db/internal/DBProcessorImpl.kt]: placeholder comment to check before merge

File: processors/db-processor/build.gradle [hunk 4]
```diff
@@ -79,6 +79,12 @@ dependencies {
     runtimeOnly project(':components:membership:certificates-service-impl')
     runtimeOnly project(':components:membership:group-policy-configuration-validation-impl')
 
+    // TODO: re-evaluate this before merge. Does this belong in the DB processor?
```
Comment (yarivus) [inline processors/db-processor/build.gradle]: placeholder comment to check before merge

File: components/interop/alias-info-read-service/src/main/kotlin/net/corda/interop/aliasinfo/read/InteropAliasInfoReadService.kt [hunk 5]
```diff
@@ -0,0 +1,8 @@
+package net.corda.interop.aliasinfo.read
+
+import net.corda.lifecycle.Lifecycle
+
+
+interface InteropAliasInfoReadService : Lifecycle {
```
Comment (yarivus) [inline components/interop/alias-info-read-service/src/main/kotlin/net/corda/interop/aliasinfo/read/InteropAliasInfoReadService.kt]: I'm not sure why you need 2 packages for InteropAliasInfoReadService & InteropAliasInfoWriteService.
You can have a single package and define both classes source files there ... 
I think it's over-engineering

Comment (yarivus) [reply components/interop/alias-info-read-service/src/main/kotlin/net/corda/interop/aliasinfo/read/InteropAliasInfoReadService.kt]: Take note this separation adds double package-info.java, detekt, build.gradle files etc

File: components/interop/alias-info-read-service-impl/src/main/kotlin/net/corda/interop/aliasinfo/read/impl/InteropAliasInfoReadServiceImpl.kt [hunk 6]
```diff
@@ -0,0 +1,45 @@
...
+import org.slf4j.LoggerFactory
+
+
+@Component(service = [InteropAliasInfoReadService::class])
+class InteropAliasInfoReadServiceImpl @Activate constructor(
+    @Reference(service = LifecycleCoordinatorFactory::class)
+    private val coordinatorFactory: LifecycleCoordinatorFactory,
+    @Reference(service = ConfigurationReadService::class)
+    private val configurationReadService: ConfigurationReadService,
+) : InteropAliasInfoReadService {
```
Comment (yarivus) [inline components/interop/alias-info-read-service-impl/src/main/kotlin/net/corda/interop/aliasinfo/read/impl/InteropAliasInfoReadServiceImpl.kt]: Maybe worth implementing InteropAliasInfoServiceImpl  which does most of the common code ...
Also, worth considering having a single InteropAliasInfoService interface

File: components/interop/alias-info-read-service-impl/src/main/kotlin/net/corda/interop/aliasinfo/read/impl/InteropAliasInfoReadServiceEventHandler.kt [hunk 7]
```diff
@@ -0,0 +1,79 @@
...
+import net.corda.lifecycle.StartEvent
+import net.corda.lifecycle.StopEvent
+import net.corda.schema.configuration.ConfigKeys
+import org.slf4j.LoggerFactory
+
+
+/**
+ * Handler for interop alias info read service lifecycle events.
+ */
+class InteropAliasInfoReadServiceEventHandler(
```
Comment (yarivus) [inline components/interop/alias-info-read-service-impl/src/main/kotlin/net/corda/interop/aliasinfo/read/impl/InteropAliasInfoReadServiceEventHandler.kt]: consider code reuse between the InfoReadServiceEventHandler and InfoWriteServiceEventHandler

**Review by yarivus — CHANGES_REQUESTED** [review]

**Comment by SaurabhR3** [comment]
This PR was just to show Julia creation of REST API, closing this PR

**Pull request closed without merging**